src/sdk/pynni/nni/compression/torch/speedup/infer_shape.py

```diff
@@ -276,7 +322,15 @@ def dropout_inshape(module_masks, mask):
     module_masks.set_output_mask(mask)
     return module_masks.output_mask
 
+def dropout_outshape(module_masks, mask):
+    if module_masks.output_mask is None:
+        module_masks.set_output_mask(mask)
+        module_masks.set_input_mask(mask)
+        return module_masks.input_mask
+    # if alreay visited
+    assert all(module_masks.output_mask.mask_index[1] == mask.mask_index[1])
 
+    return module_masks.output_mask
 
 def cat_inshape(module_masks, mask, cat_info, last_visited):
```

Comment:
Shall we also support the `cat_outshape` in the future?

Reply:
Sure.